src/signalrclient/connection_impl.cpp

```diff
@@ -85,173 +94,177 @@ namespace signalr
+                    {
+                        connection->m_logger.log(trace_level::errors,
+                            std::string("connection could not be started due to: ")
+                            .append(e.what()));
+                    }
+                    connection->change_state(connection_state::disconnected);
+                    callback(exception);
+                    return;
+                }
 
-            bool foundWebsockets = false;
-            for (auto availableTransport : negotiation_response.availableTransports)
-            {
-                if (availableTransport.transport == "WebSockets")
+                if (!response.error.empty())
                 {
-                    foundWebsockets = true;
-                    break;
+                    connection->change_state(connection_state::disconnected);
+                    callback(std::make_exception_ptr(signalr_exception(response.error)));
+                    return;
                 }
-            }
 
-            if (!foundWebsockets)
-            {
-                return pplx::task_from_exception<void>(signalr_exception("The server does not support WebSockets which is currently the only transport supported by this client."));
-            }
+                if (!response.url.empty())
+                {
+                    if (!response.accessToken.empty())
+                    {
+                        auto headers = connection->m_signalr_client_config.get_http_headers();
+                        headers[_XPLATSTR("Authorization")] = utility::conversions::to_string_t("Bearer " + response.accessToken);
+                        connection->m_signalr_client_config.set_http_headers(headers);
+                    }
+                    connection->start_negotiate(response.url, redirect_count + 1, callback);
+                    return;
+                }
 
-            // TODO: use transfer format
+                connection->m_connection_id = std::move(response.connectionId);
 
-            if (token.is_canceled())
-            {
-                pplx::cancel_current_task();
-                return pplx::task_from_result();
-            }
+                // TODO: fallback logic
 
-            return connection->start_transport(url)
-                .then([weak_connection, token](std::shared_ptr<transport> transport)
-            {
-                if (token.is_canceled())
-                {
-                    pplx::cancel_current_task();
-                    return pplx::task_from_result();
-				}
-                auto connection = weak_connection.lock();
-                if (!connection)
+                bool foundWebsockets = false;
+                for (auto& availableTransport : response.availableTransports)
                 {
-                    return pplx::task_from_exception<void>("connection no longer exists");
+                    case_insensitive_equals comparer;
+                    if (comparer(availableTransport.transport, "WebSockets"))
```

Comment:
Is availableTransport a list of strings for all available transports, or is it only one transport.